controllers/datafoundation.go

```diff
@@ -110,6 +119,14 @@ func dfIsReadyToBeRemoved(r *ManagedFusionOfferingReconciler, offering *v1alpha1
 }
 
 func dfReconcile(offeringReconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) (ctrl.Result, error) {
+	// Check if the required CRDs are installed or not
+	if !doesCRDExistsAtStart {
+		// we only check for storagecluster CRD because other crds are installed as a dependency via ocs-operator
+		if utils.MapCRDAvailability(storageClusterCRD)[storageClusterCRD] {
+			offeringReconciler.Log.Info("operator needs to restart to register watchers for offering CRDs")
+			os.Exit(1)
```

Comment:
Don't exist from within an offering code. Return a predefined error that the outer reconcile code could translate into  an `os.Exist`